docs/algorithms/SMAWK Algorithm/SMAWK Algorithm.md

```diff
@@ -0,0 +1,95 @@
+---
+
+id: SMAWK-Algorithm  
+sidebar_position: 19  
+title: "SMAWK Algorithm"  
+sidebar_label: SMAWK Algorithm  
+
+---
+
+### Definition
+
+The **SMAWK Algorithm** is a specialized algorithm for efficiently finding row minima in a certain class of totally monotone matrices. It is primarily used in problems involving dynamic programming on matrices and is an optimization over the brute-force approach of finding the minimum in each row independently.
+
+### Characteristics
+
+- **Algorithm Type**: Dynamic Programming Optimization, Matrix Algorithms.
+- **Main Operation**: Finds the minimum element in each row of a totally monotone matrix.
+- **Data Structures**: A vector to store the minimum values and another to track the active set of columns.
+- **Output**: A vector containing the minimum value of each row in the matrix.
+
+### Time Complexity
+
+- **Average Case**: \( O(n \log n) \), where \( n \) is the number of rows in the matrix.
+- **Worst Case**: \( O(n m) \), where \( m \) is the number of columns in the matrix.
+
```

Comment:
replace `\( O(n \log n) \)` with `$O(n \log n) $`, `\( n \)`  with `$n$` , `\( O(n m) \)` with `$O(m n)$`, and `\( m \)` with `$m$`